.github/workflows/ctest.yml

```diff
@@ -1,90 +1,87 @@
-name: matrix (ubuntu, macos, windows) ctest
-
+name: matrix (ubuntu, macos, windows) ctes
 on:
   push:
     paths-ignore:
       - docs/**
-      - '**.md'
-      - '**.markdown'
-    branches: [ master ]
+      - "**.md"
+      - "**.markdown"
+    branches:
+      - master
   pull_request:
     paths-ignore:
       - docs/**
-      - '**.md'
-      - '**.markdown'
-    branches: [ master ]
-      
+      - "**.md"
+      - "**.markdown"
+    branches:
+      - master
 env:
-  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
   BUILD_TYPE: Release
-  
-# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow
-# if you have to do a quick fix and the CI is already running your new commit will kill already 
-# going processes before starting new ones
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
-  
 jobs:
   build:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
     runs-on: ${{ matrix.os }}
-    
-    # The CMake configure and build commands are platform agnostic and should work equally
-    # well on Windows or Mac.  You can convert this to a matrix build if you need
-    # cross-platform coverage.
-    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
-    #runs-on: ubuntu-latest
-    #runs-on: ${{ matrix.os }}
-
     steps:
-    - uses: actions/checkout@v3
-
-    - name: Run Linux Build
-      if: matrix.os == 'ubuntu-latest'
-      run: echo "Ubuntu Latest" > release_ubuntu
-      
-    - name: Run Mac Build
-      if: matrix.os == 'macos-latest'
-      run: echo "MacOS Latest" > release_mac
-
-    - name: Run Windows Build
-      if: matrix.os == 'windows-latest'
-      run: echo "Windows Latest" > release_windows
+      # checkout full depth of history. 
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
git fetch-debth was needed or else the git history is not available and calculating the version in the cmake build won't work.